src/Http/Http.Abstractions/src/HttpContext.cs

```diff
@@ -16,6 +16,11 @@ namespace Microsoft.AspNetCore.Http;
 [DebuggerTypeProxy(typeof(HttpContextDebugView))]
 public abstract class HttpContext
 {
+    /// <summary>
+    /// Gets or sets the <see cref="Endpoint"/> for this request.
+    /// </summary>
+    public abstract Endpoint? Endpoint { get; set; }
```

Comment:
Is staying in sync with the feature part of the contract?  If so, it should be in the doc comment.

Reply:
Thank you for the feedback. I've reviewed the implementation, and since HttpContext.Endpoint is effectively a pass-through to IEndpointFeature.Endpoint, they will always stay in sync as part of the design. I’ve updated the documentation to reflect this behavior.

Reply:
We can't add new abstract members, this is a breaking change.

Reply:
> We can't add new abstract members, this is a breaking change.

 hi @davidfowl I understand that adding new abstract members can be a breaking change, and I’d like to better understand why it applies in this particular case. Since HttpContext.Endpoint is essentially a pass-through to IEndpointFeature.Endpoint, my intent was to maintain backward compatibility by ensuring that they stay in sync.

I’ve also updated the documentation to reflect this behavior and ensured that the feature is part of the design rather than introducing any new abstract functionality.

Could you please clarify further why this is considered a breaking change? Also, if there’s an alternative approach you recommend that would prevent this from being a breaking change, I’d greatly appreciate your guidance.

Reply:
It needs to be virtual. Adding new required members to an abstract class is a breaking change for any class that implements it.

https://learn.microsoft.com/en-us/dotnet/core/compatibility/library-change-rules

Reply:
> It needs to be virtual. Adding new required members to an abstract class is a breaking change for any class that implements it.
> 
> https://learn.microsoft.com/en-us/dotnet/core/compatibility/library-change-rules

Hi @davidfowl,

Thanks for the clarification. I've updated HttpContext.Endpoint to be virtual to avoid the breaking change. Please let me know if there are any further concerns